src/main_app/scheduler.py

```
@@ -0,0 +1,94 @@
+
+def collect_main_files_job() -> None:
+    """Scheduled job to collect main files."""
+    logger.info("Starting scheduled collect_main_files job")
+    try:
+        job_id = jobs_worker.start_job(user=None, job_type="collect_main_files")
+        logger.info(f"Scheduled collect_main_files job started with ID {job_id}")
+    except Exception:
+        logger.exception("Failed to start scheduled collect_main_files job")
+
+
+def init_scheduler(app: Flask) -> BackgroundScheduler | None:
```

Ruff (0.15.2)

[warning] Unused function argument: `app`

(ARG001)